extension/src/popup/components/sendPayment/SendAmount/index.tsx

```diff
@@ -65,6 +66,41 @@ const ConversionRate = ({
   </div>
 );
 
+const BalanceOption = ({
+  balance: [key, balance],
+}: {
+  balance: [string, Types.AssetBalance | Types.NativeBalance];
+}) => {
+  const [assetDomain, setAssetDomain] = useState("stellar.org");
+  const assetIssuer = "issuer" in balance.token ? balance.token.issuer.key : "";
+  const networkDetails = useSelector(settingsNetworkDetailsSelector);
+  const server = new StellarSdk.Server(networkDetails.networkUrl);
+
+  useEffect(() => {
+    const fetchAssetDomain = async () => {
+      let homeDomain = "";
+      // https://github.com/stellar/freighter/issues/410
+      try {
+        ({ home_domain: homeDomain } = await server.loadAccount(assetIssuer));
+      } catch (e) {
+        console.error(e);
+      }
+
+      setAssetDomain(homeDomain);
```

Comment:
We should be cautious displaying the domain from the asset issuers Stellar account home domain field, because anyone can put any domain into their accounts home domain. For example, I could put stellar.org in my personal domain. Or a wallet may decide to put the services domain into the home domain field of all their users.

Do we verify that the Stellar account that has the domain in the home domain field is mentioned as a `CURRENCY` in the stellar.toml of the domain to confirm there's a two-way relationship?

We started defining a protocol for this type of asset domain verification in [SEP-32](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0032.md) although it is just a proposal at this stage.

Reply:
Thanks @leighmcculloch this is a good callout. We don't do any type of verification now, but issue #410 will be a follow up to clean up this type of home_domain fetching. I'll make sure to add that there